src/Tribe/Views/V2/View.php

```diff
@@ -806,12 +819,18 @@ public function prev_url( $canonical = false, array $passthru_vars = [] ) {
 		}
 
 		// Make sure the view slug is always set to correctly match rewrites.
-		$url = add_query_arg( [ 'eventDisplay' => $this->slug ], $url );
+		$query_args['eventDisplay'] = $this->slug;
 
-		$url = $prev_page->count() > 0 ? add_query_arg( $query_args, $url ) : '';
+		if ( $prev_page->count() > 0 ) {
+			$query_args = array_merge( $query_args, $page_query_args );
+		} else {
+			$query_args = [];
+			$url        = '';
+		}
 
 		if ( ! empty( $url ) && $paged === 1 ) {
 			$url = remove_query_arg( $this->page_key, $url );
+			unset( $query_args[ $this->page_key ] );
```

Comment:
:no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 6eedb1df61eda7b4cbe043e68ff8c060</sub>